core/src/main/java/org/elasticsearch/action/fieldstats/FieldStatsRequest.java

```diff
@@ -33,17 +40,111 @@
 
     public final static String DEFAULT_LEVEL = "cluster";
 
-    private String[] fields;
+    private String[] fields = Strings.EMPTY_ARRAY;
     private String level = DEFAULT_LEVEL;
+    private IndexConstraint[] indexConstraints = new IndexConstraint[0];
 
     public String[] fields() {
         return fields;
     }
 
     public void fields(String[] fields) {
+        if (fields == null) {
+            throw new IllegalArgumentException("Specified fields can't be null");
+        }
         this.fields = fields;
     }
 
+    public IndexConstraint[] indexConstraints() {
+        return indexConstraints;
+    }
+
+    public void indexConstraints(IndexConstraint[] indexConstraints) {
```

Comment:
lets do that, but then I'll change the other methods into setters and getters too.